[BUG] FastQC AUR build fails

This is regarding the [fastq AUR](https://aur.archlinux.org/packages/fastqc) which is based on the following file and is maintained by @starsareintherose :

https://github.com/BioArchLinux/Packages/blob/master/BioArchLinux/fastqc/PKGBUILD

While updating the fastQC AUR, I got the following error. I believe it is bevause the `javac -target` should be `1.8` instead of `11` which is [defined in the `build.xml`](https://github.com/s-andrews/FastQC/blob/d4f2e91d67b765867b59cde3c6a9a6c7de34ae37/build.xml#L10). The following is verbatim from the `man javac`:

> ```
>        -target version
>               Generates class files that target a specified release of the virtual machine. Class files will
>               run on the specified target and on later releases, but not on earlier releases of the JVM.
>               Valid targets are 1.1, 1.2, 1.3, 1.4, 1.5 (also 5), 1.6 (also 6), 1.7 (also 7), and 1.8 (also 8).
>               The default for the -target option depends on the value of the -source option:
>               
>               • If the -source option is not specified, then the value of the -target option is 1.8
>               • If the -source option is 1.2, then the value of the -target option is 1.4
>               • If the -source option is 1.3, then the value of the -target option is 1.4
>               • If the -source option is 1.5, then the value of the -target option is 1.8
>               • If the -source option is 1.6, then the value of the -target is option 1.8
>               • If the -source option is 1.7, then the value of the -target is option 1.8
>               • For all other values of the -source option, the value of the -target option is the value
>                 of the -source option.
> ```



This means we have to patch the `build.xml` during the AURs `prepare` function to replace the following lines to have the correct value:

https://github.com/s-andrews/FastQC/blob/d4f2e91d67b765867b59cde3c6a9a6c7de34ae37/build.xml#L10-L11


**Log of the bug**

<details><summary>My terminal output</summary>

```console
❯  paru
:: Synchronizing package databases...
 core is up to date
 extra is up to date
 community is up to date
 multilib is up to date
:: Starting full system upgrade...
warning: r: ignoring package upgrade (4.0.3-2 => 4.2.3-1)
 there is nothing to do
:: Looking for AUR upgrades...
:: Looking for devel upgrades...
:: Resolving dependencies...
:: Calculating conflicts...
:: Calculating inner conflicts...

Aur (1) fastqc-0.12.1-2

:: Proceed to review? [Y/n]: y

:: Downloading PKGBUILDs...
 PKGBUILDs up to date
 nothing new to review
fetching devel info...
==> Making package: fastqc 0.12.1-2 (Wed 26 Apr 2023 10:47:14 AM EEST)
==> Retrieving sources...
  -> Found fastqc-0.12.1
  -> Found fastqc.patch
==> Validating source files with md5sums...
    fastqc-0.12.1 ... Passed
    fastqc.patch ... Passed
==> Making package: fastqc 0.12.1-2 (Wed 26 Apr 2023 10:47:15 AM EEST)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Found fastqc-0.12.1
  -> Found fastqc.patch
==> Validating source files with md5sums...
    fastqc-0.12.1 ... Passed
    fastqc.patch ... Passed
==> Removing existing $srcdir/ directory...
==> Extracting sources...
  -> Extracting fastqc-0.12.1 with bsdtar
==> Starting prepare()...
patching file build.xml
==> Sources are ready.
fastqc-0.12.1-2: parsing pkg list...
==> Making package: fastqc 0.12.1-2 (Wed 26 Apr 2023 10:47:16 AM EEST)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> WARNING: Using existing $srcdir/ tree
==> Removing existing $pkgdir/ directory...
==> Starting build()...
Buildfile: /home/mehrad/.cache/paru/clone/fastqc/src/FastQC-0.12.1/build.xml

build-subprojects:

init:
    [mkdir] Created dir: /home/mehrad/.cache/paru/clone/fastqc/src/FastQC-0.12.1/bin
     [copy] Copying 59 files to /home/mehrad/.cache/paru/clone/fastqc/src/FastQC-0.12.1/bin

build-project:
     [echo] FastQC: /home/mehrad/.cache/paru/clone/fastqc/src/FastQC-0.12.1/build.xml
    [javac] Compiling 138 source files to /home/mehrad/.cache/paru/clone/fastqc/src/FastQC-0.12.1/bin
    [javac] javac: invalid target release: 11
    [javac] Usage: javac <options> <source files>
    [javac] use -help for a list of possible options

BUILD FAILED
/home/mehrad/.cache/paru/clone/fastqc/src/FastQC-0.12.1/build.xml:36: Compile failed; see the compiler error output for details.

Total time: 0 seconds
==> ERROR: A failure occurred in build().
    Aborting...
error: failed to build 'fastqc-0.12.1-2':
error: packages failed to build: fastqc-0.12.1-2
```
 
 </details>

**Desktop (please complete the following information):**
 - Desktop Enviroment: KDE Plasma v5.27.4
 - Linux Kernel: 6.1.25-1-MANJARO
 - Package Version: [e.g. 22]

**Additional context**
--not applicable--

i am not the maintainer of the said package. that said, fastqc seems to have been successfully built by bioarchlinux's build server in a clean chroot as per the [log](https://build.bioarchlinux.org/api/pkg/fastqc/log/1682437849)

```
==> Starting build()...
Buildfile: /build/fastqc/src/FastQC-0.12.1/build.xml

build-subprojects:

init:
    [mkdir] Created dir: /build/fastqc/src/FastQC-0.12.1/bin
     [copy] Copying 59 files to /build/fastqc/src/FastQC-0.12.1/bin

build-project:
     [echo] FastQC: /build/fastqc/src/FastQC-0.12.1/build.xml
    [javac] Compiling 138 source files to /build/fastqc/src/FastQC-0.12.1/bin
    [javac] warning: [options] system modules path not set in conjunction with -source 11
    [javac] /build/fastqc/src/FastQC-0.12.1/org/apache/commons/math3/util/MathUtils.java:57: warning: [removal] Double(double) in Double has been deprecated and marked for removal
    [javac]         return new Double(value).hashCode();
    [javac]                ^
    [javac] /build/fastqc/src/FastQC-0.12.1/org/apache/commons/math3/util/ResizableDoubleArray.java:915: warning: [removal] Float(float) in Float has been deprecated and marked for removal
    [javac]         hashData[0] = new Float(expansionFactor).hashCode();
    [javac]                       ^
    [javac] /build/fastqc/src/FastQC-0.12.1/org/apache/commons/math3/util/ResizableDoubleArray.java:916: warning: [removal] Float(float) in Float has been deprecated and marked for removal
    [javac]         hashData[1] = new Float(contractionCriteria).hashCode();
    [javac]                       ^
    [javac] Note: Some input files use or override a deprecated API.
    [javac] Note: Recompile with -Xlint:deprecation for details.
    [javac] Note: Some input files use unchecked or unsafe operations.
    [javac] Note: Recompile with -Xlint:unchecked for details.
    [javac] 4 warnings

build:

BUILD SUCCESSFUL

```
if i may make a suggestion why not add Bioarchlinux repo to your pacman and repository and install fastqc from there?

--- a/BioArchLinux/fastqc/PKGBUILD
+++ b/BioArchLinux/fastqc/PKGBUILD
@@ -7,27 +7,30 @@ pkgver=0.12.1
 pkgrel=3
 pkgdesc='A quality control tool for high throughput sequence data.'
 arch=('any')
-url="http://www.bioinformatics.babraham.ac.uk/projects/fastqc"
+url="https://www.bioinformatics.babraham.ac.uk/projects/fastqc"
 license=('GPL3')
 depends=('perl' 'java-runtime>=11' 'ttf-dejavu')
 makedepends=('ant' 'java-environment>=11')
-source=("$pkgname-$pkgver::https://github.com/s-andrews/FastQC/archive/refs/tags/v${pkgver}.tar.gz"
-	"$pkgname.patch::https://github.com/s-andrews/FastQC/commit/e73f094ec165882cc71707a5dd5d3dd263a51b83.patch")
-md5sums=('2d22a29649394f589e6f03d8d8c3eec9'
-         '37a54b16eeb73d39a88893e326192e29')
+source=("$pkgname-$pkgver.tar.gz::https://github.com/s-andrews/FastQC/archive/refs/tags/v${pkgver}.tar.gz"
+        "fastqc.sh"
+        "$pkgname.patch::https://github.com/s-andrews/FastQC/commit/e73f094ec165882cc71707a5dd5d3dd263a51b83.patch")
+sha256sums=('cad8bfd11ecf388204c28a62b162dda032aac8ecd08819c49b053aa66613e92b'
+            'e1b8e3d5ef1b5804af65f888edbd89c98018b8b43fc114e7cc6bada741c9a352'
+            '61cdeadcc3f2885a4aa5412ea2b215364f279ffc5aecec513afa2bc00c182c15')
 prepare() {
   cd $_pkgname-$pkgver
-  patch -p1 < $srcdir/$pkgname.patch
+  patch -p1 -i "../$pkgname.patch"
 }
 build() {
   cd $_pkgname-$pkgver
-  ant
+  # use latest available java
+  local _javaenv="$(archlinux-java status | tail -n +2 | cut -d ' ' -f 3 | sort -nk 2 -t - | tail -n 1)"
+  JAVA_HOME="/usr/lib/jvm/$_javaenv" ant
 }
 package() {
-  mkdir -p $pkgdir/usr/share/$pkgname/
-  cp -r $srcdir/$_pkgname-$pkgver/bin/* $pkgdir/usr/share/$pkgname/
+  install -d "$pkgdir/usr/share/$pkgname"
+  cp -r "$_pkgname-$pkgver/bin"/* "$pkgdir/usr/share/$pkgname/"
+  chmod 755 "$pkgdir/usr/share/$pkgname/fastqc"
 
-  mkdir -p "${pkgdir}/usr/bin/"
-  chmod 755 $pkgdir/usr/share/$pkgname/fastqc
-  ln -s "/usr/share/$pkgname/fastqc" "${pkgdir}/usr/bin/"
+  install -Dm755 fastqc.sh "$pkgdir/usr/bin/fastqc"
 }

--- a/BioArchLinux/fastqc/fastqc.sh
+++ b/BioArchLinux/fastqc/fastqc.sh
@@ -1,0 +1,15 @@
+#!/bin/sh
+
+if [ -n "$FASTQC_JAVA" ]; then
+  javaenv="$FASTQC_JAVA"
+else
+  default_java="$(archlinux-java get)"
+  if [ "$(echo "$default_java" | cut -d - -f 2)" -ge 11 ]; then
+    javaenv="$default_java"
+  else
+    # use latest available java
+    javaenv="$(archlinux-java status | tail -n +2 | cut -d ' ' -f 3 | sort -nk 2 -t - | tail -n 1)"
+  fi
+fi
+
+PATH="/usr/lib/jvm/$javaenv/bin:$PATH" exec /usr/share/fastqc/fastqc "$@"

